fineract-provider/src/main/java/org/apache/fineract/infrastructure/core/config/FineractModeValidationConfig.java

```diff
@@ -0,0 +1,47 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.fineract.infrastructure.core.config;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.BeansException;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@Component
+@ConditionalOnExpression("#{ systemEnvironment['FINERACT_MODE_READ_ENABLED'] == false and systemEnvironment['FINERACT_MODE_WRITE_ENABLED'] == false and systemEnvironment['FINERACT_MODE_BATCH_ENABLED'] == false}")
+public class FineractModeValidationConfig implements ApplicationContextAware {
```

Comment:
I think we need an `@Order` annotation with the highest order to make sure Fineract is not going to do anything if it's none of the instances, for example no Liquibase migration runs.
@taskain7 @vidakovic thoughts?

Reply:
@galovics ... you are right... the order could be a concern here... of course we wouldn't want Liquibase accidentally kicking in even though the instance was intended as read-only. Makes sense to use `@Order` here @taskain7